server/chat-plugins/calculator.ts

```diff
@@ -132,17 +139,51 @@ export const commands: ChatCommands = {
 	math: "calculate",
 	calculate(target, room, user) {
 		if (!target) return this.parse('/help calculate');
+		let base = 10;
+		if (target.includes('0x')) {
+			base = 16;
+		} else if (target.includes('0o')) {
+			base = 8;
+		} else if (target.includes('0b')) {
+			base = 2;
+		}
+
+		const baseMatchResult = target.match(/\bin\s+([a-zA-Z]+)\b/);
+		if (baseMatchResult) {
+			switch (toID(baseMatchResult[1])) {
+			case 'decimal': case 'dec': base = 10; break;
+			case 'hexadecimal': case 'hex': base = 16; break;
+			case 'octal': case 'oct': base = 8; break;
+			case 'binary': case 'bin': base = 2; break;
+			default:
+				return this.errorReply(`Unrecognized base "${baseMatchResult[1]}". Valid options are binary or bin, octal or oct, decimal or dec, and hexadecimal or hex.`);
+			}
+		}
+		const expression = target.replace(/\bin\s+([a-zA-Z]+)\b/g, '').trim();
+
 		if (!this.runBroadcast()) return;
 		try {
-			const result = solveRPN(parseMathematicalExpression(target));
-			this.sendReplyBox(Chat.html`${target}<br />= <strong>${Chat.stringify(result)}</strong>`);
+			const result = solveRPN(parseMathematicalExpression(expression));
+			let baseResult = ``;
+			if (result) {
+				baseResult = `${BASE_PREFIXES[base]}${result.toString(base).toUpperCase()}`;
+				if (baseResult === result.toString()) baseResult = '';
+			}
+			let resultStr = '';
+			if (baseResult) {
+				resultStr = `<strong>${baseResult}</strong> = ${result}`;
+			} else {
+				resultStr = `<strong>${result}</strong>`;
+			}
+			this.sendReplyBox(`${expression}<br />= ${resultStr}`);
 		} catch (e) {
 			this.sendReplyBox(
-				Chat.html`${target}<br />= <span class="message-error"><strong>Invalid input:</strong> ${e.message}</span>`
+				Chat.html`${expression}<br />= <span class="message-error"><strong>Invalid input:</strong> ${e.message}</span>`
 			);
 		}
 	},
 	calculatehelp: [
-		`/calculate [arithmetical question] - Calculates an arithmetical question. Supports PEMDAS (Parenthesis, Exponents, Multiplication, Division, Addition and Subtraction), pi and e.`,
+		`/calculate [arithmetical question] [in [base]] - Calculates an arithmetical question. Supports PEMDAS (Parenthesis, Exponents, Multiplication, Division, Addition and Subtraction), pi and e.
```

Comment:
```suggestion
		`/calculate [arithmetical question] [in [base]] - Calculates an arithmetical question. Supports PEMDAS (Parenthesis, Exponents, Multiplication, Division, Addition and Subtraction), pi and e.
		`/calculate [arithmetical question] in [base] - Returns the result in a specific base. [base] can be bin, oct, or hex.
```

Reply:
I'm not a native speaker, but "to" feels weird to me in "10 + 12 to hex". I guess it implies a verb like "change base to" or something? But since with singular numbers I'd always say "this is in hex" or "this is in binary", that feels more natural to me.

Reply:
To me, `to` feels more natural; the implied verb is `converted to`. They're both grammatically correct, but `in` feels slightly more awkward. I don't mind either one, if you feel strongly.